### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by samarthdd**
Integration of malicious file check
Integration of minio adapter

### CODE REVIEW

**Comment by belwase** [comment]
This branch is already included in this PR : https://github.com/k8-proxy/k8-test-data/pull/27

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
 
+    def get_sites_to_run(self, scrape_site=None):
         if scrape_site is None:
             section = "SCRAPE_SITE"
             try:
                 config_obj = ConfigReader(section).read_config()
                 sites_arr = config_obj['scrape_sites'].split(',')
-                print(sites_arr)
             except KeyError as ke:
                 raise CloseSpider(reason="Error while getting site list from config.")
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/glasswall_crawler_runner.py:38]: On Exception no logging.

File: upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/pipelines.py [hunk 2]
```diff
@@ -5,9 +5,53 @@
...
+import os
 
+from scrapy.pipelines.files import FilesPipeline
+from scrapy.utils.misc import md5sum
+from six import BytesIO
 
-class MaliciousFileCrawlerPipeline:
-    def process_item(self, item, spider):
-        return item
+from .constants import base_unzip_path, zip_download_path
```
Comment (susilupwork) [inline upwork-devs/samarth_dd/gw_crawler/malicious_file_crawler/src/pipelines.py:15]: pipelines.py : you are you are storing the response.body to buf but why no validation and exception handling, response.body may give you 404 response, no validation done.

**Review by susilupwork — COMMENTED** [review]
Repeat issue : All file and network operation code should have exception handling.
Commenting is not sufficient and not proper.  Your virus_total.py is having good comments. check that. 
No logging in minio_adapter.py
You should not check in log file. 
Are you using nose test or pytest, if not used remove from requirements.txt
Improve unit tests

**Pull request closed without merging**